src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -3043,18 +3202,25 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 			return NULL;
 
 		MonoClass *arg_class = mono_class_from_mono_type_internal (fsig->params [0]);
+		int esize;
 
 		if (fsig->params [0]->type == MONO_TYPE_GENERICINST) {
 			MonoType *etype = mono_class_get_context (arg_class)->class_inst->type_argv [0];
-			int size = mono_class_value_size (arg_class, NULL);
-			int esize = mono_class_value_size (mono_class_from_mono_type_internal (etype), NULL);
-			elems = size / esize;
+			esize = mono_class_value_size (mono_class_from_mono_type_internal (etype), NULL);
 		} else {
 			// This exists to handle the static extension methods for Vector2/3/4, Quaternion, and Plane
 			// which live on System.Numerics.Vector
 
 			arg0_type = MONO_TYPE_R4;
-			elems = 4;
+			esize = 4;
+		}
+
+		int size = mono_class_value_size (arg_class, NULL);
+		elems = size / esize;
+
+		if (size != 16) {
```

Comment:
Same questions as above

Reply:
Same as above, the `WithElement` APIs for Vector2/3 are new as of this PR.

The blocker here is the same as well, there needs to be a centralized helper that ensures the handling of Vector2/3 is done correctly to avoid the invalid IR shape otherwise being created.